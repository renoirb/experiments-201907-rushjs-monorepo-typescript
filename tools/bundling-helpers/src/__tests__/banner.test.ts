/* eslint-env jest */

import {
  PeopleField,
} from '../people-field'
import {
  PackageJson,
} from '../package'
import {
  createBannerInfo,
  createBannerFooter,
} from '../banner'

const peopleFixture: PeopleField = {
  email: '7.9@adjunct.3.unimatrix.1.borg',
  name: 'Seven Of Nine',
}

const packageFixture: PackageJson = {
  name: 'assimilate',
  version: '436.0.1',
}

describe('createBannerInfo', () => {
  it('Snapshot sample', () => {
    const subject = createBannerInfo(packageFixture)
    expect(subject).toMatchSnapshot()
  })

  it('Provides default author, license, copyright fields', () => {
    const subject = createBannerInfo(packageFixture)
    expect(subject).toHaveProperty('author', 'ACME Corp.')
    expect(subject).toHaveProperty(
      'copyright',
      'Copyright (c) 2003-' + new Date().getFullYear() + ' ACME Corp.',
    )
    expect(subject).toHaveProperty('license', 'LicenseRef-LICENSE')
    expect(subject).toHaveProperty('name', 'assimilate')
    expect(subject).toHaveProperty('vendor', 'ACME Corp.')
  })

  it('Supports defining own author', () => {
    const subject = createBannerInfo({
      ...packageFixture,
      author: { ...peopleFixture },
    })
    expect(subject).toMatchSnapshot()
    expect(subject).toHaveProperty(
      'author',
      'Seven Of Nine <7.9@adjunct.3.unimatrix.1.borg>',
    )
  })

  it('Supports defining own license', () => {
    const subject = createBannerInfo({
      ...packageFixture,
      license: 'MIT',
    })
    expect(subject).toMatchSnapshot()
    expect(subject).toHaveProperty('license', 'MIT')
  })
})

describe('createBannerFooter', () => {
  it('Snapshot sample', () => {
    const seed = createBannerInfo(packageFixture)
    const subject = createBannerFooter(seed)
    expect(subject).toMatchSnapshot()
  })

  it('Is possible to add arbitrary lines', () => {
    const append = ['target: node', 'browserslist: node 10']
    const seed = createBannerInfo(packageFixture)
    const subject = createBannerFooter(seed, append)
    const YYYY = new Date().getFullYear()
    expect(subject).toHaveProperty(
      'banner',
      `/**
 * assimilate v436.0.1
 *
 * Maintainer: ACME Corp.
 *
 * target: node
 * browserslist: node 10
 *
 * LicenseRef-LICENSE
 *
 * Â© 2003-${YYYY} ACME Corp.
 */`,
    )
    expect(subject).toHaveProperty('footer', `/** ACME Corp.  */`)
  })

  it('Is possible to override default vendor', () => {
    const vendor = 'Gammick International 2000 Inc.'
    const seed = createBannerInfo(packageFixture, vendor)
    const subject = createBannerFooter(seed)
    expect(subject).toHaveProperty('footer', `/** ${vendor}  */`)
  })
})
