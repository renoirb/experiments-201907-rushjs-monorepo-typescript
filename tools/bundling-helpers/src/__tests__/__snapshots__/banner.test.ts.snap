// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBannerFooter Snapshot sample 1`] = `
Object {
  "banner": "/**
 * assimilate v436.0.1
 *
 * Maintainer: ACME Corp.
 *
 * LicenseRef-LICENSE
 *
 * Â© 2003-2020 ACME Corp.
 */",
  "footer": "/** ACME Corp.  */",
}
`;

exports[`createBannerInfo Snapshot sample 1`] = `
Object {
  "author": "ACME Corp.",
  "copyright": "Copyright (c) 2003-2020 ACME Corp.",
  "firstYear": 2003,
  "license": "LicenseRef-LICENSE",
  "name": "assimilate",
  "vendor": "ACME Corp.",
  "version": "436.0.1",
}
`;

exports[`createBannerInfo Supports defining own author 1`] = `
Object {
  "author": "Seven Of Nine <7.9@adjunct.3.unimatrix.1.borg>",
  "copyright": "Copyright (c) 2003-2020 ACME Corp.",
  "firstYear": 2003,
  "license": "LicenseRef-LICENSE",
  "name": "assimilate",
  "vendor": "ACME Corp.",
  "version": "436.0.1",
}
`;

exports[`createBannerInfo Supports defining own license 1`] = `
Object {
  "author": "ACME Corp.",
  "copyright": "Copyright (c) 2003-2020 ACME Corp.",
  "firstYear": 2003,
  "license": "MIT",
  "name": "assimilate",
  "vendor": "ACME Corp.",
  "version": "436.0.1",
}
`;
